src/ipa_pc/mod.rs

```diff
@@ -1286,15 +1379,15 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
                 hiding_bound = std::cmp::max(hiding_bound, cur_poly.hiding_bound());
                 poly += (*coeff, cur_poly.polynomial());
 
-                combined_rand += &(cur_rand.rand * coeff);
+                combined_rand += &(cur_rand.randomness().rand * coeff);
                 combined_shifted_rand = Self::combine_shifted_rand(
                     combined_shifted_rand,
-                    cur_rand.shifted_rand,
+                    cur_rand.randomness().shifted_rand,
                     *coeff,
                 );
 
                 let commitment = cur_comm.commitment();
-                combined_comm += &commitment.comm.mul(*coeff);
+                combined_comm += &commitment.comm[0].mul(*coeff);
```

Comment:
If we use segmentation, we need combine all segment commitments here (and not just comm[0]).

Reply:
Done